src/main/java/com/exasol/sql/expression/rendering/ValueExpressionRenderer.java

```diff
@@ -1,127 +1,239 @@
 package com.exasol.sql.expression.rendering;
 
+import java.util.List;
+
 import com.exasol.datatype.type.DataType;
 import com.exasol.sql.ColumnsDefinition;
 import com.exasol.sql.UnnamedPlaceholder;
 import com.exasol.sql.expression.*;
+import com.exasol.sql.expression.comparison.Comparison;
+import com.exasol.sql.expression.comparison.ComparisonVisitor;
+import com.exasol.sql.expression.comparison.LikeComparison;
+import com.exasol.sql.expression.comparison.SimpleComparison;
+import com.exasol.sql.expression.function.AbstractFunction;
 import com.exasol.sql.expression.function.Function;
+import com.exasol.sql.expression.function.FunctionVisitor;
 import com.exasol.sql.expression.function.exasol.CastExasolFunction;
 import com.exasol.sql.expression.function.exasol.ExasolFunction;
 import com.exasol.sql.expression.function.exasol.ExasolUdf;
+import com.exasol.sql.expression.literal.*;
 import com.exasol.sql.rendering.ColumnsDefinitionRenderer;
 import com.exasol.sql.rendering.StringRendererConfig;
 
 /**
  * Renderer for common value expressions.
  */
-public class ValueExpressionRenderer extends AbstractExpressionRenderer implements ValueExpressionVisitor {
+public class ValueExpressionRenderer extends AbstractExpressionRenderer implements BooleanExpressionVisitor,
+        ComparisonVisitor, ValueExpressionVisitor, LiteralVisitor, FunctionVisitor {
+    int nestedLevel = 0;
+
+    /**
+     * Create a new instance of {@link ValueExpressionRenderer}.
+     * 
+     * @param config render configuration
+     */
     public ValueExpressionRenderer(final StringRendererConfig config) {
         super(config);
     }
 
     @Override
-    public void visit(final StringLiteral literal) {
-        appendCommaWhenNeeded(literal);
-        append("'");
-        append(literal.toString());
-        append("'");
-        setLastVisited(literal);
+    public void visit(final Not not) {
+        appendKeyword("NOT");
+        startParenthesis();
+        not.getOperand().accept((BooleanExpressionVisitor) this);
+        endParenthesis();
     }
 
     @Override
-    public void visit(final IntegerLiteral literal) {
-        appendCommaWhenNeeded(literal);
-        append(literal.toString());
-        setLastVisited(literal);
+    public void visit(final And and) {
+        startParenthesisIfNested();
+        append(" AND ", and.getOperands());
```

Comment:
If you use a literal, make sure that rendering configuration is applied. If I look at this code, I doubt that the case rendering will be handled correctly. Please add a test case with lower case config to verify.

Reply:
`append(" AND ", and.getOperands())` `uses appendKeyword(separator);` internaly

tested in TestBooleanExpressionRenderer#testAndWhitNestedOrInLowercase

Reply:
OK.